### DISCUSSION TITLE
How to embed gl-function in raylib framwork?why the following program can not get the dash line?

### DISCUSSION THREAD

**Discussion by aliule2020** [raysan5/raylib#2475] [Q&A] [answered]
#define APIENTRY __stdcall
#define WINGDIAPI __declspec(dllimport)

#include "raylib.h"
#include "imgui.h"
#include "imgui_impl_opengl3.h"
#include "imgui_impl_raylib.h"
#include <gl/GL.h>
#include "rlgl.h"

void DrawDash(int sx,int sy,int ex,int ey,Color co)
{                        // Reset current matrix (MODELVIEW)

    //glLineStipple(1, 0x0101);

    rlDrawRenderBatchActive();
    rlBegin(RL_LINES);
    rlColor4f(co.r, co.g, co.b, co.a);
    rlVertex2f(sx, sy);
    rlVertex2f(ex, ey);
    rlEnd();

}

int main(int argc, char* argv[])
{
    int screenWidth = 1024;
    int screenHeight = 768;

    InitWindow(screenWidth, screenHeight, "raylib-imgui");
    SetWindowState(FLAG_WINDOW_RESIZABLE);

    SetTargetFPS(60);

    ImGui::CreateContext();
    ImGui::StyleColorsDark();

    ImGui_ImplOpenGL3_Init();
    ImGui_ImplRaylib_Init();

    

    while (!WindowShouldClose())
    {
        ImGui_ImplOpenGL3_NewFrame();
        ImGui_ImplRaylib_NewFrame();
        ImGui::NewFrame();
        ImGui_ImplRaylib_ProcessEvent();

        
        BeginDrawing();

        {
            ClearBackground(RAYWHITE);
          
            DrawCircle(100, 100, 50,RED);

            **DrawDash(200, 200, 350, 350, BLUE);**    //here it dosen't work

            ImGui::ShowDemoWindow();

            ImGui::Render();
            ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
        }

        EndDrawing();
    }

    ImGui_ImplRaylib_Shutdown();
    ImGui_ImplOpenGL3_Shutdown();

    CloseWindow();

    return 0;
}

**Comment by raysan5** [comment, marked as answer]
@aliule2020 Here it is an example of OpenGL interoperativility: https://github.com/raysan5/raylib/blob/master/examples/others/raylib_opengl_interop.c

But I'm afraid `glLineStipple()` was deprecated in OpenGL 3.1, you should recompile raylib for OpenGL 2.1 or OpenGL 1.1.

**Reply by aliule2020** [reply]
Thank you

#include "raylib.h"
#include "rlgl.h"

void DrawDash(float sx,float sy,float ex,float ey,int segnum,float segratio, Color color, float width)
{
    float mx;
    float my;
    float lenx = (ex - sx) / segnum;
    float leny = (ey - sy) / segnum;
    mx = sx;
    my = sy;

    rlBegin(RL_LINES);
    rlSetLineWidth(width);
    for (int i = 0; i < segnum; i++)
    {
        rlColor4ub (color.r, color.g, color.b, color.a); 
        rlVertex3f(mx, my, 0);

        rlColor4ub(~color.r, ~color.g, ~color.b, color.a);
        rlVertex3f(mx + lenx * segratio, my + leny * segratio, 0);

        mx = mx + lenx;
        my = my + leny;
    }
    rlEnd();

}


int main()
{
    const int screenWidth = 800;
    const int screenHeight = 800;
    InitWindow(screenWidth, screenHeight, "raylib DashLine");

    while (!WindowShouldClose())
    {
        BeginDrawing();
        ClearBackground(RAYWHITE);

           DrawDash(100, 100, 600, 600, 20, 0.5, Color{ 0 , 255 ,0 ,255 }, 10);

        EndDrawing();
    }

    CloseWindow();

}

In raylib4.2, work very well.
